Merge ver/1.16 into master

Proximyst opened the pull request:
There should first be a proper look through to make sure we comment shit with Papercut where necessary.

Proximyst committed:
Initial 1.16 support

This dropped the implementation for the StructureGenerateEvent.

Proximyst committed:
Reimplement StructureGenerateEvent

Proximyst committed:
1.16 dimensions for world config

Proximyst committed:
Fix #isCollidable error

Proximyst committed:
Boolean flip

Proximyst committed:
Redo item serialization & add entity ser API

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
069fa98d Merge pull request #3639 from kickash32/1.16-updates
e94688ca Fix comment mistake
527dd9f1 fixup
5958abd6 delete ported patches
2192ff1b fix headers
501c1c78 port 4 patches
52f85036 someone left this in?
989d44eb organize patches

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
20e92f90 Disable the memory information in /tps unless argument is passed
d871ae80 Fix missing chunks on client (View distance issue)
15613201 Merge pull request #3642 from kickash32/1.16-updates
c0d07c1b Updated Upstream (Bukkit/CraftBukkit/Spigot)
74bf22ad more patch organization
78c5491f Fix Piglin issue
68aa46cc Remove unneeded patch
986ebc68 Updated Upstream (CraftBukkit)

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
86ee6c96 Log Minecraft Internal Scheduler logs correctly
e4d10c3c Revert recent changes around player skulls using user cache
bf4b5415 Fix missing hunk from ProfileWhitelistVerifyEvent (Fixes #3645)
b6925c36 Remove no longer needed undead horse leash patch
5e9cc3a2 Fix null pointer in portal search config
3a77cce3 Fix Per World Difficulty / Remembering Difficulty

Proximyst committed:
Kick players before disabling plugins

Proximyst committed:
Fake seed & trample changes

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
778d63b1 Add Gradle usage information to README (#3700)
25c14f76 Fix compile
78c85523 Updated Upstream (CraftBukkit)
8a4af269 fix EntityDismountEvent being fired on gen threads
55ab552c Don't fire unleash/dismount events from worldgen threads
9edbfe5f Support old UUID format for NBT
fb772923 Improve Legacy Component serialization size
8806d0a8 Fix case where some skulls still have string UUID's
9073cfa4 Updated Upstream (Bukkit/CraftBukkit)
35282e46 Fix Reloadable save players error on stop
bcc619ae revert chat changes for now until can get it properly working
04d09e1d Add /paper dumpitem command
3703b177 Fix texture profile properties getting duplicated - Fixes #3667
4f455872 Don't allow null UUID's for chat
f4b7b588 Improve Legacy Component serialization size
f339d6e1 Fix more null pointer issues around deleted worlds - Fixes #3660
6c6318cb Fix player skulls rendering - Fixes #3672
e928b165 [1.16] Fix MobGoals#getAllGoals not actually returning all goals (#3670)
358ea3f7 [CI-SKIP] Change 1.16 branch to show 1.16.1 API in readme (#3664)
943c9238 Fix NPE from an invalid spawn dimension (Fixes #3660)
b27469dc [CI-SKIP] Drop accidental import from last commit
4da9f70a Restore Player#refreshPlayer (Fixes #3650)

Proximyst committed:
Log villager removals with the API

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
ab347c4c9 Updated Upstream (Bukkit/CraftBukkit/Spigot)
e2e09326d [Auto] Updated Upstream (Bukkit/CraftBukkit)

Proximyst committed:
Prevent overflow in villager gossip

Proximyst committed:
Patch PaperMC/Paper#3708

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
d78cbb855 [1.16] Remove some streams from structures (#3704)
8522e7b27 Fix shutdown on eula not being agreed (Fixes #4647)
c43dcbfcf Improve old UUID support, clean up old legacy keys.
f63e3dd13 Allow teleporting through end gateways as a passenger/vehicle
b7d9e3d5d Convert legacy attributes in Item Meta
4fd0d54ee Fix server not stopping when eula not agreed yet
6f30ecdeb Clean up duplicated GameProfile Properties
d30472260 Fix supporting more places of the old UUID format
f5265d668 Updated Upstream (Bukkit/CraftBukkit/Spigot)
7ce690f49 [Auto] Updated Upstream (Bukkit/CraftBukkit)
bcab622e5 [Auto] Updated Upstream (CraftBukkit)

Proximyst committed:
Make Entity#isCollidable non-final

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
4b5b20d4c Fix hex colours not being allowed in MOTD (#3755)
f32d7fb2d Add more material tags for 1.16 (#3750)
a96d3e898 [Auto] Updated Upstream (CraftBukkit)
0823d3fdc [Auto] Updated Upstream (Bukkit/CraftBukkit)
75645fbfb [Auto] Updated Upstream (CraftBukkit)

Proximyst committed:
Fix MOTD new line support

Proximyst committed:
Fully async advancement loading

Proximyst committed:
Villagers are always persistent

Proximyst committed:
Fix timings export for config

Proximyst committed:
Ensure papercut comments

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
fcbeac8a9 [CI-SKIP] Readme update (#3702)
824f8086c Bandaid italic legacy serialization #3757 (#3760)
1eeb8ad14 Support components in ItemMeta (#3513)
ea8823ef5 Fix MOTD new line support (#3756)

NahuLD commented on patches/api/0009-Add-Raw-Byte-Entity-Serialization.patch:
```diff
@@ -57,6 +59,5 @@ index ae1ebc5f8a61165d95184e2a12a270825ed1fd2d..83eddbdabddcf4db726cb210de12c679
-     // Paper end
- 
-     // Papercut start
+     // Papercut end
```
Is the start of this wrapper in bundled with another patch?

Proximyst replied:
It's bundled with the raw methods patch, but it shouldn't add the end here, I'll fix that later.

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
a12a67785 Support String UUID's in old player skulls
b3f265d1b Updated Upstream (Bukkit/CraftBukkit/Spigot)

Proximyst committed:
Updated Upstream (Paper)

Upstream has released updates that appears to apply and compile correctly

Paper Changes:
4fd174452 Spawn player in correct world on login
43eb073db [Auto] Updated Upstream (Bukkit/Spigot)
b25236f0b [Auto] Updated Upstream (CraftBukkit)

Proximyst committed:
Re-do patch: prevent overflow in villager gossip

Proximyst committed:
Add dragon block protections

Proximyst committed:
Invalid data? More like no data!
